### PR TITLE
airport_challenge

### PR DESCRIPTION
**Pull request by sherwingp**
Todo:
- [x] Write tests for edge cases
- [ ] Refactor

User stories:

- [x] As an air traffic controller 
        So I can get passengers to a destination 
        I want to instruct a plane to land at an airport

- [x] As an air traffic controller 
        So I can get passengers on the way to their destination 
        I want to instruct a plane to take off from an airport and confirm that it is no longer in the airport

- [x] As an air traffic controller 
        To ensure safety 
        I want to prevent landing when the airport is full 

- [x] As the system designer
        So that the software can be used for many different airports
        I would like a default airport capacity that can be overridden as appropriate

- [x] As an air traffic controller 
            To ensure safety 
            I want to prevent takeoff when weather is stormy 

- [x] As an air traffic controller 
        To ensure safety 
        I want to prevent landing when weather is stormy

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Airport
+  attr_reader :planes, :capacity, :weather
+  DEFAULT_CAPACITY = 10
+  WEATHER_SCALE = 10
+
+  def initialize(capacity = 10)
```
Comment (tc1316) [inline lib/airport.rb:6]: Could use capacity = DEFAULT_CAPACITY

File: lib/airport.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    @planes = []
+    @capacity = capacity
+    @weather = rand(WEATHER_SCALE)
+  end
+
+  def full?
+    return @planes.length >= DEFAULT_CAPACITY
```
Comment (tc1316) [inline lib/airport.rb:13]: Dont need @planes since its been declared in attr_reader

File: lib/airport.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def stormy?
+    @weather >= WEATHER_SCALE
```
Comment (tc1316) [inline lib/airport.rb:17]: Likewise above

File: lib/plane.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+class Plane
+  attr_reader :flying
+  def initialize
+    @flying = true
+  end
+
+  def land(airport)
+    fail "airport is full" if airport.full?
+    fail "weather is stormy" if airport.stormy?    
+    fail "plane has already landed" if flying? == false
```
Comment (tc1316) [inline lib/plane.rb:10]: Good use of guard conditions

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+describe Airport do
+  subject(:airport) { Airport.new }
+
+  it 'default capacity can be overridden' do
+    airport = Airport.new(3)
+    expect(airport.capacity).to be(3)
+  end
+
+  it 'can check for stormy weather' do
+    expect(subject.stormy?).to be_truthy.or be_falsy
```
Comment (tc1316) [inline spec/airport_spec.rb:13]: Cool!

File: spec/plane_spec.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+describe Plane do
+  subject(:plane) { Plane.new }
+
+  context 'landing' do
+    it 'lands at airport' do
+      airport = Airport.new
+      allow(airport).to receive(:stormy?) { false }
+      subject.land(airport)
+      expect(airport.planes.length).to be(1)
```
Comment (tc1316) [inline spec/plane_spec.rb:12]: Alternative suggestion: expect(airport.planes).to include(plane)

File: spec/plane_spec.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    end
+
+    it 'cannot land if airport is full' do
+      airport = Airport.new
+      Airport::DEFAULT_CAPACITY.times { airport.planes.push("plane") }
```
Comment (tc1316) [inline spec/plane_spec.rb:17]: Good use of calling DEFAULT_CAPACITY here to show that the airport will be full

File: spec/plane_spec.rb [hunk 8]
```diff
@@ -0,0 +1,61 @@
+require 'plane'
+require 'airport'
+
+describe Plane do
+  subject(:plane) { Plane.new }
+
+  context 'landing' do
```
Comment (tc1316) [inline spec/plane_spec.rb:7]: Context use makes tests clearer to read

File: spec/plane_spec.rb [hunk 9]
```diff
@@ -0,0 +1,61 @@
+require 'plane'
+require 'airport'
+
+describe Plane do
+  subject(:plane) { Plane.new }
+
```
Comment (tc1316) [inline spec/plane_spec.rb:6]: You could create the airport up here instead of in each test

**Review by tc1316 — CHANGES_REQUESTED** [review]
Code review from Titus

File: lib/airport.rb (hunk 2 above)
Comment (sherwingp) [reply lib/airport.rb:13]: Thanks Titus! I didn't know that!

File: lib/plane.rb (hunk 4 above)
Comment (sherwingp) [reply lib/plane.rb:10]: Thanks Titus!

**Pull request closed without merging**